src/main/webapp/app/quiz/participate/quiz.component.ts

```diff
@@ -775,48 +776,81 @@ export class QuizComponent implements OnInit, OnDestroy {
         }
     }
 
+    /**
+     * Checks if the array is empty for the question based on the id
+     */
+    checkForSelectedAnswerOption(): boolean {
```

Comment:
First, the name of the function is confusing. It handles MC and DnD questions so why is it called 'checkForSelectedAnswerOption'? -> better would be to call it something like checkForAnsweredQuestions.

In addition this method only returns a boolean value if the answer array has elements, it does not return anything if the length is 0. This is rather bad, because I would assume that I definitely get a return value here.

I think it would be much better if you introduce a map<questionId, boolean> in which you store whether the student has worked on a particular question. In the beginning, the map should be initialized with only false values, and later it is filled with true, when the student has worked on a particular question.

Reply:
Renamed the function and added a second return type